src/common/ConsoleMessage.ts

```diff
@@ -16,20 +16,55 @@
 
 import { JSHandle } from './JSHandle';
 
-interface ConsoleMessageLocation {
+/**
+ * @public
+ */
+export interface ConsoleMessageLocation {
+  /**
+   * URL of the resource if known or `undefined` otherwise.
+   */
   url?: string;
+
+  /**
+   * 0-based line number in the resource if known or `undefined` otherwise.
+   */
   lineNumber?: number;
+
+  /**
+   * 0-based column number in the resource if known or `undefined` otherwise.
+   */
   columnNumber?: number;
 }
 
+// Prettier seems to struggle with the ConsoleMessageType declaration
+// so it is switched off just for that block.
+/* eslint-disable prettier/prettier */
+
+/**
+ * The supported types for console messages.
+ */
+export type ConsoleMessageType = 'log' | 'debug' | 'info' | 'error' | 'warning' |
+    'dir' | 'dirxml' | 'table' | 'trace' | 'clear' | 'startGroup' |
+    'startGroupCollapsed' | 'endGroup' | 'assert' | 'profile' |
+    'profileEnd' | 'count' | 'timeEnd' | 'verbose';
```

Comment:
does this type already exist in `src/protocol.d.ts` ? If so could we import it to avoid the duplication? It might not (there's ongoing work to get us onto the devtools-protocol npm package) and in that case this is fine

Reply:
Done by extracting an existing one from Protocol.